### PR TITLE
Update README and add HPC/Jupyter Guidance

### LINKED ISSUE #330: [Doc]: Add some guidance for xcdat and HPC/Jupyter + Extend README
### Describe your documentation update

It would be useful to provide some general guidance to HPC and Jupyter users. We could also more clearly link to readthedocs and other resources in the README file.

### PR DESCRIPTION
**Pull request by pochedls**
## Description

Initial proposal for documentation update (update readme and provide some guidance for HPC+Jupyter). 

- Closes #330 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream modules

If applicable:

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass with my changes (locally and CI/CD build)
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have noted that this is a breaking change for a major release (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Review by lee1043 — APPROVED** [review]

**Comment by chengzhuzhang** [comment]
@pochedls what you have is pretty good. I just tried to remove some redundancy in `getting-started` and `HPC/Jupyter` session.  

One thing I'm not clear is about jupyterlab, ipython and ipykernel.  I included jupyterlab as an optional  package, because it is also included in the xcdat_test env. Not  sure if jupyterlab  covers  ipython and  ipykernel. I'm hoping  to have both guide to be consistent in terms of suggested optional packages. @tomvothecoder and @pochedls any insights?

**Review by tomvothecoder — APPROVED** [review]
Hi @pochedls, the guide is looking good. 

I made some additional minor tweaks in this [commit](https://github.com/xCDAT/xcdat/pull/331/commits/3951e2ba431a39e9ede4eb87627881ed237f2523). Please review the commit message and the related diffs, then proceed with merging if you are in agreement. Otherwise, you can edit as needed and merge.

**Comment by tomvothecoder** [comment]
> One thing I'm not clear is about jupyterlab, ipython and ipykernel. I included jupyterlab as an optional package, because it is also included in the xcdat_test env. Not sure if jupyterlab covers ipython and ipykernel. I'm hoping to have both guide to be consistent in terms of suggested optional packages. @tomvothecoder and @pochedls any insights?

`jupyterlab` (which is intended on replacing `notebook`) include `ipykernel` and `ipython`. 

`ipykernel` is the Python execution backend for Jupyter. Installing only `ipykernel` means you won't get the front-end Jupyter interfaces included in `jupyterlab`/`notebook`. This is fine for environments that already provide `jupyterlab`/`notebook`.

File: README.rst [hunk 1]
```diff
@@ -49,8 +55,17 @@ A major design philosophy of xCDAT is streamlining the user experience while dev
...
 .. _CDAT: https://github.com/CDAT/cdat
 
+Getting Started
+---------------
+
+The best resource for getting started with xCDAT is via our `documentation page <https://xcdat.readthedocs.io/en/latest/>`__. There, we provide guidance for setting up your environment on your `computer <https://xcdat.readthedocs.io/en/latest/getting-started.html>`_ generally or on `HPC/Jupyter <https://xcdat.readthedocs.io/en/latest/getting-started-hpc-jupyter.html>`_ environment. We also include an `API Overview <https://xcdat.readthedocs.io/en/latest/api.html>`_ and `Gallery <https://xcdat.readthedocs.io/en/latest/gallery.html>`_ to highlight xCDAT functionality.
+
+xCDAT invites discussion on version releases, architecture, new features suggestions, and other topics on the `GitHub discussion <https://github.com/xCDAT/xcdat/discussions>`_ page. Users and contributors can also view and open issues on our `GitHub Issue Tracker <https://github.com/xCDAT/xcdat/issues>`_.
+
+We welcome and appreciate contributions to xCDAT. If you'd like to help improve xCDAT, please checkout our `Contributing Guide <https://xcdat.readthedocs.io/en/latest/contributing.html>`_.
```
Comment (tomvothecoder) [inline README.rst:65]: I forgot to format this section here and in `index.rst` to 88 characters (same as Python code) for readability. Not a high priority though and other `.rst` files need the same update.

**Comment by chengzhuzhang** [comment]
Thank you @tomvothecoder ! This helps me a lot. when searching online, I found below chart which is also useful for understanding the whole JupyterLab system...

image

**Review by chengzhuzhang — APPROVED** [review]

**Comment by pochedls** [comment]
I made a couple minor changes, but I think this could be merged.

**Review by chengzhuzhang — APPROVED** [review]
All looks good. Please feel free to merge, Steve. @pochedls 
Thank you everyone for working on this PR...

**Pull request merged**